Crash on search

![](http://i.imgur.com/1Ij883d.png)

cUrl query:

```
curl 'https://kinto-ota.dev.mozaws.net/v1/buckets/build-hub/collections/releases/search' -H 'Origin: http://localhost:8000' -H 'Accept-Encoding: gzip, deflate, br' -H 'Accept-Language: en' -H 'User-Agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/58.0.3029.110 Safari/537.36' -H 'Content-Type: application/json' -H 'Accept: */*' -H 'Referer: http://localhost:8000/' -H 'Connection: keep-alive' -H 'DNT: 1' --data-binary '{"size":0,"query":{"bool":{"must":[]}},"aggregations":{"product_filters":{"source.product":{"terms":{"field":"source.product"}}},"channel_filters":{"target.channel":{"terms":{"field":"target.channel"}}},"platform_filters":{"target.platform":{"terms":{"field":"target.platform"}}},"version_filters":{"target.version":{"terms":{"field":"target.version"}}},"locale_filters":{"target.locale":{"terms":{"field":"target.locale"}}}}}' --compressed
```

```
    response = view(request)
  File "/home/ubuntu/venvs/kinto/local/lib/python3.5/site-packages/cornice/service.py", line 493, in wrapper
    response = view_(request)
  File "/home/ubuntu/venvs/kinto/local/lib/python3.5/site-packages/kinto_elasticsearch/views.py", line 63, in post_search        
    return search_view(request, body=body)
  File "/home/ubuntu/venvs/kinto/local/lib/python3.5/site-packages/kinto_elasticsearch/views.py", line 44, in search_view        
    message = e.info["error"]["reason"]
TypeError: string indices must be integers
[pid: 16075|app: 0|req: 12183/58769] 172.31.13.135 () {54 vars in 904 bytes} [Tue Jun 13 09:58:11 2017] POST /v1/buckets/build-hub/collections/releases/search => generated 177 bytes in 41 msecs (HTTP/1.1 500) 5 headers in 232 bytes (2 switches on core 0)
```


`Could not find aggregator type [source.product]`

```
Caused by: org.elasticsearch.search.SearchParseException: [kinto-build-hub-releases][1]: query[*:*],from[-1],size[0]: Parse Failure [Could not find aggregator type [source.product] in [product_filters]]
        at org.elasticsearch.search.aggregations.AggregatorParsers.parseAggregators(AggregatorParsers.java:146)
        at org.elasticsearch.search.aggregations.AggregatorParsers.parseAggregators(AggregatorParsers.java:78)
        at org.elasticsearch.search.aggregations.AggregationParseElement.parse(AggregationParseElement.java:60)
        at org.elasticsearch.search.SearchService.parseSource(SearchService.java:731)
        ... 9 more
```

--- a/CHANGELOG.rst
+++ b/CHANGELOG.rst
@@ -8,6 +8,8 @@ Changelog
 **Bug fixes**
 
 - Limit the number of results returned by default (fixes #45)
+- Fix crash on search parse exceptions (fixes #44)
+
 
 0.1.0 (2017-05-26)
 ------------------

--- a/kinto_elasticsearch/views.py
+++ b/kinto_elasticsearch/views.py
@@ -48,8 +48,12 @@ def search_view(request, **kwargs):
 
     except elasticsearch.RequestError as e:
         # Malformed query.
-        message = e.info["error"]["reason"]
-        details = e.info["error"]["root_cause"][0]
+        if isinstance(e.info["error"], dict):
+            message = e.info["error"]["reason"]
+            details = e.info["error"]["root_cause"][0]
+        else:
+            message = e.info["error"]
+            details = None
         response = http_error(httpexceptions.HTTPBadRequest(),
                               errno=ERRORS.INVALID_PARAMETERS,
                               message=message,

--- a/tests/test_elasticsearch.py
+++ b/tests/test_elasticsearch.py
@@ -384,3 +384,11 @@ class SchemaSupport(BaseWebTest, unittest.TestCase):
             {"key": "abc", "doc_count": 1},
             {"key": "efg", "doc_count": 1},
         ]
+
+    def test_search_parse_exception_returns_400(self):
+        e = elasticsearch.RequestError('', '',
+                                       {"error": "Could not find aggregator type"})
+        with mock.patch("kinto_elasticsearch.indexer.Indexer.search",
+                        side_effect=e):
+            self.app.post_json("/buckets/bid/collections/cid/search",
+                               headers=self.headers, status=400)

